packages/hardhat-core/test/internal/hardhat-network/provider/modules/hardhat.ts

```diff
@@ -365,6 +374,737 @@ describe("Hardhat module", function () {
+            numberToRpcQuantity(targetNonce),
+          ]);
+
+          // Act 2: Mine a block
+          await this.provider.send("evm_mine");
+
+          // Assert: Ensure modified nonce has persisted.
+          const resultingNonce = await this.provider.send(
+            "eth_getTransactionCount",
+            [DEFAULT_ACCOUNTS_ADDRESSES[0], currentBlockNumber]
+          );
+          assert.equal(resultingNonce, targetNonce);
+        });
+      });
+
+      describe("hardhat_setStorageSlot", function () {
+        it("should reject an invalid address", async function () {
+          await assertInvalidArgumentsError(
+            this.provider,
+            "hardhat_setStorageSlot",
+            ["0x1234", numberToRpcQuantity(0), numberToRpcQuantity(99)],
+            'Errors encountered in param 0: Invalid value "0x1234" supplied to : ADDRESS'
+          );
+        });
+
+        it("should reject storage key that is non-numeric", async function () {
+          await assertInvalidArgumentsError(
+            this.provider,
+            "hardhat_setStorageSlot",
+            [DEFAULT_ACCOUNTS_ADDRESSES[0], "xyz", numberToRpcQuantity(99)],
+            'Errors encountered in param 1: Invalid value "xyz" supplied to : QUANTITY'
+          );
+        });
+
+        it("should reject a storage key that is greater than 32 bytes", async function () {
+          const MAX_WORD_VALUE = new BN(2).pow(new BN(256));
+          await assertInvalidInputError(
+            this.provider,
+            "hardhat_setStorageSlot",
+            [
+              DEFAULT_ACCOUNTS_ADDRESSES[0],
+              numberToRpcQuantity(MAX_WORD_VALUE.add(new BN(1))),
+              "0xff",
+            ],
+            "Storage key must not be greater than 2^256. Received 115792089237316195423570985008687907853269984665640564039457584007913129639937."
+          );
+        });
+
+        for (const badInputLength of [1, 2, 31, 33, 64]) {
+          it(`should reject a value that is ${badInputLength} (not exactly 32) bytes long`, async function () {
+            await assertInvalidInputError(
+              this.provider,
+              "hardhat_setStorageSlot",
+              [
+                DEFAULT_ACCOUNTS_ADDRESSES[0],
+                numberToRpcQuantity(0),
+                `0x${"ff".repeat(badInputLength)}`,
+              ],
+              `Storage value must be exactly 32 bytes long. Received 0x${"ff".repeat(
+                badInputLength
+              )}, which is ${badInputLength} bytes long.`
+            );
+          });
+        }
+
+        it("should not reject valid argument types", async function () {
+          await this.provider.send("hardhat_setStorageSlot", [
+            DEFAULT_ACCOUNTS_ADDRESSES[0],
+            numberToRpcQuantity(0),
+            `0x${"ff".repeat(32)}`,
+          ]);
+        });
+
+        it("should result in modified storage", async function () {
+          const targetStorageSlot = 1;
```

Comment:
This variable doesn't seem to be used (the other variables called `targetStorageSlot` seem unused too).

Reply:
Thanks for noticing this. I wish there was a linter rule for this.  Addressed in https://github.com/nomiclabs/hardhat/pull/1435/commits/4d5b673e39e93d1f14e7ef40da599280bc106d95